examples/roitberg/roitberg.py

```diff
@@ -15,71 +21,118 @@ def convert_species_to_atomic_nums(s):
     res.append(PERIODIC_TABLE[k])
   return np.array(res, dtype=np.float32)
 
+def path(*dir_segments):
+  return os.path.abspath(os.path.expanduser(os.path.join(*dir_segments)))
+
+def path_not_empty(d):
+  d = path(d)
+  return os.path.exists(d) and len(os.listdir(d)) > 1
+
+def find_or_create_path(*dir_segments):
+  """
+  Pretty much what it sounds like: finds a directory path or creates one if it
+  doesn't exist.
+
+  Also adds in the functionality of os.path.join, for extra helpfulness: the
+  dir_segments will be joined together, then checked for existence.
+  """
+  d = path(*dir_segments)
+
+  if not os.path.exists(d):
+    os.makedirs(d, exist_ok=True)
+
+  return d
+
+def setup_work_dirs(args):
+  """
+  Sets up the work directory structure beneath the directory in +args.work_dir+.
+
+  If args.work_dir is not a valid directory, errors out with a warning to the user,
+  on the theory that the program shouldn't be chucking lots of data in places the
+  user isn't explicitly aware of.
+
+  Below args.work_dir, creates new subdirectories with impunity. The various work
+  directories will be set in +args+ for later reference.
+
+  Parameters:
+    args - the result of calling ArgumentParser.parse_args()
+  """
+  args.work_dir = path(args.work_dir)
+
+  if not os.path.exists(args.work_dir):
+    # we don't create the work dir if it doesn't already exist:
+    # don't want to silently dump big files on the user's FS without
+    # their knowledge.
+    sys.stderr.write("Work directory '%s' does not exist. "
+                     "Create it, or specify a new directory with the -w option.\n" % args.work_dir)
+    sys.exit(-1)
+  elif args.clean_work_dir:
+    # python doesn't have an easy way to remove all files in a directory
+    # this will *probably* work, but might fail on weird setups.
+    shutil.rmtree(args.work_dir)
```

Comment:
do you also want ignore_errors=True? Not sure what happens if the work_dir doesn't exist and we attempt to clean.

Reply:
Fair point. I handled that in the makedirs call on the next line, but not this one.

Reply:
oh, actually, I remember why: if it fails because of file permissions or something, I didn't just
want to blindly proceed. it's possible to handle this with a callback, but it seemed excessive.